gum/guminterceptor.c

```diff
@@ -535,6 +536,9 @@ gum_interceptor_replace (GumInterceptor * self,
 
   function_ctx->replacement_data = replacement_data;
   function_ctx->replacement_function = replacement_function;
+  if (original_function != NULL) {
+    original_function = function_ctx->on_invoke_trampoline;
```

Comment:
This should be `*original_function =` so that the caller's variable is updated. This code changes where `original_function` points rather than the value of the caller's pointer.

Reply:
Yup that's what I did originally then when adding the if I accidentally dropped it. Thanks